### PR TITLE
FIX Makes sure score is passed to ReduceLROnPlateau

### PR DESCRIPTION
**Pull request by thomasjpfan**
Closes https://github.com/skorch-dev/skorch/issues/363

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - CLI helper function now also supports normal (i.e. non-skorch) sklearn estimators
 - Disabling all callbacks is now supported (which allows reducing overhead,
   which is especially relevant for small models).
+- `LRScheduler` now currently passes the score/loss in the history to `ReduceLROnPlateau`. (#738)
```
Comment (BenjaminBossan) [inline CHANGES.md:16]: Not sure if I understand this sentence correctly, maybe change the wording?

File: skorch/callbacks/lr_scheduler.py [hunk 2]
```diff
@@ -149,12 +149,7 @@ def on_epoch_end(self, net, **kwargs):
             if callable(self.monitor):
                 score = self.monitor(net)
             else:
-                if self.lr_scheduler_.mode == 'max':
-                    score = -np.inf
-                elif self.lr_scheduler_.mode == 'min':
-                    score = np.inf
-                else:
-                    score = net.history[-1, self.monitor]
+                score = net.history[-1, self.monitor]
```
Comment (BenjaminBossan) [inline skorch/callbacks/lr_scheduler.py:152]: What do you think: We could catch the `KeyError` and augment the error message by telling the user that they may possibly need to change the order of callbacks if they reference a metric that was not yet computed?

**Review by BenjaminBossan — APPROVED** [review]
Thanks for taking this. PR LGTM, just two minor comments.

File: skorch/callbacks/lr_scheduler.py (hunk 2 above)
Comment (thomasjpfan) [reply skorch/callbacks/lr_scheduler.py]: I think its a good idea. I updated the PR to include this nicer error message.

File: skorch/callbacks/lr_scheduler.py [hunk 3]
```diff
@@ -149,7 +149,14 @@ def on_epoch_end(self, net, **kwargs):
             if callable(self.monitor):
                 score = self.monitor(net)
             else:
-                score = net.history[-1, self.monitor]
+                try:
+                    score = net.history[-1, self.monitor]
+                except KeyError:
+                    raise ValueError(
+                        f"'{self.monitor}' was not found in history. A "
+                        f"PassthroughScoring callback with name='{self.monitor}' "
```
Comment (BenjaminBossan) [inline skorch/callbacks/lr_scheduler.py:157]:
```suggestion
                        f"Scoring callback with name='{self.monitor}' "
```

I would suggest to be more agnostic about which kind of scoring is used, since `EpochScoring` would work as well.

File: skorch/callbacks/lr_scheduler.py [hunk 4]
```diff
@@ -149,12 +149,14 @@ def on_epoch_end(self, net, **kwargs):
...
                     score = net.history[-1, self.monitor]
+                except KeyError as e:
+                    raise ValueError(
+                        f"'{self.monitor}' was not found in history. A "
+                        f"PassthroughScoring callback with name='{self.monitor}' "
+                        "should be placed before the LRScheduler callback"
+                    ) from e
 
             self.lr_scheduler_.step(score)
             # ReduceLROnPlateau does not expose the current lr so it can't be recorded
```
Comment (jiajiexiao) [inline skorch/callbacks/lr_scheduler.py:162]: Thanks for the fix in this PR, which I found while troubleshooting the same unexpected behavior as mentioned in the issue https://github.com/skorch-dev/skorch/issues/363. 

I noticed this comment but I guess how the learning rate was updated should be tracked in ``net.history`` even for `ReduceLROnPlateau`. As being discussed in https://discuss.pytorch.org/t/how-to-retrieve-learning-rate-from-reducelronplateau-scheduler/54234/3 and https://github.com/pytorch/pytorch/issues/2829, I would suggest replacing the comment line [162](https://github.com/skorch-dev/skorch/blob/cadb46914203cde74fcf08af0565cde8e059b0c0/skorch/callbacks/lr_scheduler.py#L162) with the following:

```
            if self.event_name is not None:
                net.history.record(self.event_name, self.lr_scheduler_._last_lr[0])
```

**Review by jiajiexiao — CHANGES_REQUESTED** [review]

File: skorch/callbacks/lr_scheduler.py (hunk 4 above)
Comment (thomasjpfan) [reply skorch/callbacks/lr_scheduler.py:162]: We can discuss this in another PR since this is a feature unrelated to the bug fix.

Comment (BenjaminBossan) [reply skorch/callbacks/lr_scheduler.py:162]: I created an issue (#739) so that we don't forget. @jiajiexiao does that reflect what you meant?

Comment (jiajiexiao) [reply skorch/callbacks/lr_scheduler.py:162]: Sure. Thanks, @thomasjpfan @BenjaminBossan
